### ISSUE TITLE
Error status registers during an operation, and otherwise

### ISSUE THREAD

**Issue by imphil** [lowRISC/opentitan#7901]
Multiple IP blocks in OpenTitan have two main *operational states*: performing an operation ("busy"), or not ("idle"). 

An *operation* is a chunk of work that is performed by the IP block. Most times (always?) an operation is triggered by the host CPU, performed by the IP block, and then a completion signal is sent back to the host CPU, e.g. in the form of an interrupt or a change in the IP block's STATUS register which software on Ibex is polling.

IP blocks in OpenTitan are able to detect errors in both operational states and have CSRs which indicate the error that was detected. Typically, two CSRs are provided: 
- one CSR which is used for errors which are observed during an operation ("synchronous errors"). This register is also read by host software to check if the operation was successful.
- another CSR for errors which are detected outside of an operation or "asynchronous errors". This register is typically used for debugging only and might end up in the crash dump that the alert handler records.

It would be nice if we could come up with shared semantics and ideally even naming for these error status registers.

We put a fair amount of thought into this topic for OTBN and have written it down at https://docs.opentitan.org/hw/ip/otbn/doc/index.html#design-details-error-handling-and-reporting. 

@tjaychen mentioned in https://github.com/lowRISC/opentitan/issues/7625#issuecomment-902885833 that keymgr has similar functionality with slightly different naming (ERR_CODE vs ERR_BITS and FAULT_STATUS vs FATAL_ALERT_CAUSE).

To push this discussion forward, I'd be interested in

* Views from other IP blocks if such an error reporting scheme would work for your IP block as well. @vogelpi @msfschaffner @tjaychen @martin-lueker @eunchan @mwbranstad 
* And of course there's opportunity to paint a shed, so please let me know your preferred register naming as well :smile: 

I'll then come up with a consolidated proposal.

**Comment by tjaychen** [comment]
This issue somewhat duplicates the idea behind #531 
There is also a [general proposal](https://docs.google.com/presentation/d/1_TjfPF2zWtUHb2IC-C2USyK7jH5u7YgnbOex_GCk7ds/edit) for how to unify errors that should be considered at a later date.

**Comment by andreaskurth** [comment]
I came across this issue while triaging `spi_host`. IIUC, this change is not necessary for M2.5 nor the other currently planned milestones. I'm thus tagging it https://github.com/lowRISC/opentitan/labels/Type%3AFutureRelease.

**Comment by vogelpi** [comment]
Sounds good to me @andreaskurth , I've triaged this for `entropy_src`.

**Comment by GregAC** [comment]
Triaged for `otbn` I agree with the above

**Comment by cdgori** [comment]
Agreed @msfschaffner - this is a "would have been nice" (and would be good if any clean-sheet work on new blocks is done in the future to align error signaling) but seems unlikely for PROD.

**Comment by vogelpi** [comment]
I agree that we won't be able to align the behavior of our blocks for production. But it's probably worth to formulate a recommendation e.g. as part of the comportability spec for future clean-sheet work. I think there are other, similarly suitable issues and I will create an new issue to collect those.

**Comment by vogelpi** [comment]
I am closing this issue as we're not going to rework existing blocks. Instead we should formulate recommendations which is tracked in #20680.
